services/github/github-checks-status.service.js

```diff
@@ -0,0 +1,99 @@
+'use strict'
+
+const Joi = require('joi')
+const { NotFound, InvalidParameter } = require('..')
+const { GithubAuthV3Service } = require('./github-auth-service')
+const { documentation, errorMessagesFor } = require('./github-helpers')
+
+const schema = Joi.object({
+  state: Joi.equal('failure', 'pending', 'success'),
+}).required()
+
+module.exports = class GithubChecksStatus extends GithubAuthV3Service {
+  static category = 'build'
+  static route = {
+    base: 'github/checks-status',
+    pattern: ':user/:repo/:ref',
+  }
+
+  static examples = [
+    {
+      title: 'GitHub branch checks state',
+      namedParams: {
+        user: 'badges',
+        repo: 'shields',
+        ref: 'master',
+      },
+      staticPreview: this.render({
+        state: 'success',
+      }),
+      keywords: ['status'],
+      documentation,
+    },
+    {
+      title: 'GitHub commit checks state',
+      namedParams: {
+        user: 'badges',
+        repo: 'shields',
+        ref: '91b108d4b7359b2f8794a4614c11cb1157dc9fff',
+      },
+      staticPreview: this.render({
+        state: 'success',
+      }),
+      keywords: ['status'],
+      documentation,
+    },
+    {
+      title: 'GitHub tag checks state',
+      namedParams: {
+        user: 'badges',
+        repo: 'shields',
+        ref: '3.3.0',
+      },
+      staticPreview: this.render({
+        state: 'success',
+      }),
+      keywords: ['status'],
+      documentation,
+    },
+  ]
+
+  static defaultBadgeData = { label: 'checks' }
+
+  static render({ state }) {
+    if (state === 'success') {
+      return {
+        message: `passing`,
+        color: 'brightgreen',
+      }
+    } else if (state === 'pending') {
+      return {
+        message: `pending`,
+        color: 'yellow',
+      }
+    } else {
+      return {
+        message: `failing`,
+        color: 'red',
+      }
+    }
+  }
+
```

Comment:
We can drop this function altogether since rendering is handled by the common util. Note that this changes the messages ever so slightly from what is here right now, but that's a desired change so that these badges are consistent with how we handle build badges

```suggestion
```